### DISCUSSION TITLE
Visualizzare l’xml della fattura elettronica associata ad una spesa

### DISCUSSION THREAD

**Discussion by dcesca** [fattureincloud/api#424] [Q&A] [answered]
Buongiorno, vorrei sapere in che modo si possa ottenere l’html/pdf o “render” dell’xml della fattura elettronica associato a una spesa. Dal portale di FIC si apre un dettaglio che renderizza in html tabellare la fattura eletronica che puó essere poi stampata/esportata in pdf, vorrei fare la stessa cosa utilizzando l’API, è possibile?

**Comment by valmoz** [comment]
Buongiorno @dcesca 
ha già visto questa pagina? [Download the E-invoice XML](https://developers.fattureincloud.it/docs/guides/e-invoice-management#arrow_down-download-the-e-invoice-xml)

Grazie
Mauro

**Reply by dcesca** [reply]
Buongiorno @valmoz,
per leggere o fare operazioni con l’xml della fattura elettronica non ho problemi. 
Non trovo il modo di visualizzare in formato html (lo stesso formato html che si ottiene se da fatture in cloud si vuole stampare appunto una fattura/spesa). In questo modo dalla mia app posso far aprire al personale direttamente la spesa da controllare prima di inviare poi la chiamata API per settarne lo stato a pagato.

Eventualmente se non è stata prevista questa possibilitá tramite API conoscete qualche libreria che faccia questa operazione?

Altrimenti devo creare io una funzione che trasformi l’xml della fattura elettronica in formato leggibile e stampabile, grazie

**Reply by valmoz** [reply]
Buongiorno @dcesca 
l'unica alternativa è scaricare il PDF della fattura (il body della Get Issued Document) ma non credo sia quello che le serve.

Per visualizzare un XML in un formato leggibile è possibile applicare un css o xsl. 
Può seguire ad esempio le modalità elencate a [questo link](https://webreference.com/xml/basics/display-xml/)

Esistono alcuni xsl già disponibili online, ad esempio sulle pagine seguenti:
- https://www.fatturapa.gov.it/it/norme-e-regole/documentazione-fattura-elettronica/formato-fatturapa/ (cerchi "Fogli di stile")
- https://for.assosoftware.it/assoinvoice 

grazie
Mauro

**Reply by dcesca** [reply]
Grazie @valmoz l'approccio con i fogli di stile mi va bene alla fine. 
Però non mi è chiaro come ottenere l'XML della spesa ricevuta (non vedo il campo come per issued document per capirci)
https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/received_documents/info

Grazie

**Reply by emajo** [reply, marked as answer]
Ciao @dcesca, l'url dell'xml si trova all'interno del campo `data.attachment_url` della risposta alla chiamata [Get Received Document](https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/received_documents/-document_id-), se si vuole portare l'utente alla visualizzazione tabellare del documento bisognerà reindirizzarlo verso l'url https://secure.fattureincloud.it/backend_apiV2/e_invoices/preview?stylesheet=assosoftware&url={attalchment_url}

Grazie,
Emanuele.
